apteryx-xml.h

```diff
@@ -89,6 +90,7 @@ char *sch_translate_from (sch_node * node, char *value);
 bool sch_validate_pattern (sch_node * node, const char *value);
 gboolean sch_match_name (const char *s1, const char *s2);
 bool sch_ns_match (sch_node *node, void *ns);
+char *sch_find_model_path_by_href_prefix (sch_instance * instance, char *href, char *prefix);
```

Comment:
i really still don't like this function ...

actually can we not just call sch_node_child_ns with NULL for the parent ... which would indicate you are getting the top level node???? then just call sch_path on the node?

Reply:
We still need a function similar to sch_find_model_path_by_href_prefix as it is doing a node lookup based on the namespace matching only. There is no child to match so we can not call _sch_node_child. I have renamed the function  and taken your suggestion of only returning the matching sch_node and using sch_path in netconf.c